vm/src/dictdatatype.rs

```diff
@@ -573,13 +606,35 @@ impl<T: Clone> Dict<T> {
         Ok(removed)
     }
 
-    pub fn pop_back(&self) -> Option<(PyObjectRef, T)> {
-        let mut inner = self.write();
-        inner.entries.pop().map(|entry| {
-            inner.used -= 1;
-            inner.indices[entry.index] = IndexEntry::DUMMY;
-            (entry.key, entry.value)
-        })
+    pub fn pop_back(&self, vm: &VirtualMachine) -> Option<(PyObjectRef, T)> {
+        let inner = self.read();
```

Comment:
this line is the cause of deadlock. this `read` lock is not removed while calling `pop_inner` which make `write` lock